core/io/mod.rs

```diff
@@ -21,21 +22,34 @@ pub trait IO {
     fn run_once(&self) -> Result<()>;
 }
 
-pub type Complete = dyn Fn(&Buffer);
+pub type Complete = dyn Fn(Rc<RefCell<Buffer>>);
 pub type WriteComplete = dyn Fn(usize);
 
-pub struct Completion {
-    pub buf: RefCell<Buffer>,
+pub enum Completion {
+    Read(ReadCompletion),
+    Write(WriteCompletion),
+}
+
+pub struct ReadCompletion {
+    pub buf: Rc<RefCell<Buffer>>,
     pub complete: Box<Complete>,
 }
 
+impl Completion {
+    pub fn complete(&self) {
+        match self {
+            Completion::Read(r) => r.complete(),
+            Completion::Write(w) => w.complete(234234), // fix
```

Comment:
I think we should get as argument the result from the Completion Queue, I believe the `result` may relate to the `written bytes`?

Reply:
Yes, that's ehy there is a //fix 😇